### PR TITLE
Drop IE11 support.

### PR DESCRIPTION
**Pull request by trusktr**
## **Summary**

Microsoft is about to do the same thing: https://docs.microsoft.com/en-us/lifecycle/products/internet-explorer-11

Plus, Microsoft already has "IE Mode" in Edge which it will support until 2029: https://www.microsoft.com/en-us/edge/business/ie-mode

There will be no reason to keep IE support soon.

## **What kind of change does this PR introduce?**

  Docs

## **For any code change,**

- [ ] Related documentation has been updated if needed
- [ ] Related tests have been updated or tests have been added

## **Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

## **Related issue, if any:**

closes https://github.com/docsifyjs/docsify/issues/1221
closes https://github.com/docsifyjs/docsify/issues/707

## **Tested in the following browsers:**

- [x] Chrome
- [x] Firefox
- [x] Safari
- [x] Edge
- [ ] IE 😄

### CODE REVIEW

**Comment by jhildenbiddle** [comment]
We have discussed and agreed to a plan for dropping support for IE11 several times in the past.  See https://github.com/docsifyjs/docsify/issues/657#issuecomment-458341229 for one of those conversations.

TL;DR: We can drop support for IE11 with v5, but we should not bump to v5 just to drop IE11 support.

We're all ready to put IE11 behind us. Given the date on the comment linked above, we've been ready for over three years to move on. That said, we can't drop support for IE11 without publishing a new major version, and because of Docsify's history of linking to CDN URLs locked to the `@latest` version the jump from v4 to v5 is not as straightforward as a typical major version bump. There are ways to work around this, but the launch of v5 will require a conscientious effort to not break what could be a significant number of live Docsify sites.

Equally important, we should not be so quick to assume legacy browsers (like IE11) are not being used anywhere. Here's just one data point to consider: I created [css-vars-ponyfill](https://github.com/jhildenbiddle/css-vars-ponyfill) to add support for CSS custom properties (aka "CSS variables") to legacy browsers which is downloaded ~130 million times per month from one CDN ([jsDelivr](https://www.jsdelivr.com/package/npm/css-vars-ponyfill)) and ~500 thousand times per month from [npm](https://www.npmjs.com/package/css-vars-ponyfill). Here are live stats:

jsDelivr hits (npm) npm

We can't derive the number of IE11 users from those stats, but the stats do indicate that many developers and site visitors continue to use tools designed specifically to support IE11 even into 2022 (as sad as that may be). I continue to get questions from developers to this day asking things like how to integrate the ponyfill into their app which is using the latest version of Angular.

For comparison, here are the same live stats for docsify (~7M/mo for [jsDelivr](https://www.jsdelivr.com/package/npm/docsify), 117K/mo for [npm](https://www.npmjs.com/package/docsify) when written):

jsDelivr hits (npm) npm

FYI: The linked issues above were fixed by #1743. I've marked them as resolved and closed them.

@docsifyjs/core @docsifyjs/reviewers: Vote to close this PR and drop IE11 support with the next major version.

**Comment by trusktr** [comment]
What if we move the a new version (whether it is 4.x, 5.x, or etc) with build output at a new path, and leave the last IE-compatible version at the same path so `@latest` with the current URLs will never break?

**Review by Koooooo-7 — COMMENTED** [review]
I agree with @jhildenbiddle  that we need not drop this IE support for now. 
If user has new request on IE 11 we could mention them we will remove IE support in the future.

**Comment by jhildenbiddle** [comment]
> What if we move the a new version (whether it is 4.x, 5.x, or etc) build output to a new path, and leave the last IE-compatible version at the same path so that `@latest` with the current URLs will never break?

That's an option. Users would lose the ability to load the latest version of Docsify using the `@latest` URL version lock or by omitting a version lock from the URL entirely, but maybe that isn't such a bad thing.

But for now...

> The TL;DR is that we can do this with a major version bump (v4 => v5), but we should not publish a new major version just to drop IE11 support.

Let's close this PR and open where we can discuss approaches for releasing v5 (hopefully with a "how should we do it" focus instead of a "what should it contain" debate).

@trusktr -- I generally prefer to let authors close their own issues/PRs, so I'm leaving this open until you're ready.

**Comment by jhildenbiddle** [comment]
Closing due to reviewer consensus and lack of inactivity. We'll revisit legacy support in v5.
